### ISSUE TITLE
firefox extention

### ISSUE THREAD

**Issue by a8underscore** [DisboxApp/web#4]
Firefox support would be great, although not really a high priority issue ig

**Comment by WolfDWyc** [comment]
Hey, thanks for opening this issue!
I agree this is a problem. 

With the current implementation, this would require support from Firefox for the [File System Access API ](https://developer.mozilla.org/en-US/docs/Web/API/File_System_Access_API), or to somehow use the [ponyfill](https://github.com/jimmywarting/native-file-system-adapter), which I've tried but couldn't get working.

When I messed with it last time, I think one of the problems I ran into with other ways to download and upload files, is that they didn't support files bigger than the memory size, and big files seemed like it's worth the trade-off. Of course, it might be a better option to simply have a different implementation for each browser, and it could also be that I'm simply wrong and there are better options.

Anyway, I've been quite busy with other things for the past few weeks, so I don't really have time to mess with it so much anymore, but if you find any implementation that works well for with big files, and is cross-platform, I'd love to test it out!

**Comment by DisboxApp** [comment]
Hey, thanks for opening this issue!
I agree this is a problem. 

With the current implementation, this would require support from Firefox for the [File System Access API ](https://developer.mozilla.org/en-US/docs/Web/API/File_System_Access_API), or to somehow use the [ponyfill](https://github.com/jimmywarting/native-file-system-adapter), which I've tried but couldn't get working.

When I messed with it last time, I think one of the problems I ran into with other ways to download and upload files, is that they didn't support files bigger than the memory size, and big files seemed like it's worth the trade-off. Of course, it might be a better option to simply have a different implementation for each browser, and it could also be that I'm simply wrong and there are better options.

Anyway, I've been quite busy with other things for the past few weeks, so I don't really have time to mess with it so much anymore, but if you find any implementation that works well for with big files, and is cross-platform, I'd love to test it out!
